From aed037e8c3222418bc1b23f2c8fb46a7b01ee912 Mon Sep 17 00:00:00 2001
From: Kent Russell <kent.russell@amd.com>
Date: Fri, 21 Oct 2016 11:52:06 -0400
Subject: [PATCH 1898/2094] drm/amdgpu: Add a wrapper for
 amdgpu_amdkfd_bo_validate

In the 4.7 kernel, the validate function for cs and gem use a specific
format with a pointer to the parameters being passed in, then the bo.
Add a wrapper to match that format to assist in reconciling the code to
the 4.7 branch.

Change-Id: I1f5d257da1562498492fce244614c09249a94169
Signed-off-by: Kent Russell <kent.russell@amd.com>
---
 drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd_gpuvm.c | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd_gpuvm.c b/drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd_gpuvm.c
index 9b3ee7e..8a1becf 100644
--- a/drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd_gpuvm.c
+++ b/drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd_gpuvm.c
@@ -54,6 +54,12 @@ struct kfd_mem_usage_limit {
 
 static struct kfd_mem_usage_limit kfd_mem_limit;
 
+/* Struct used for amdgpu_amdkfd_bo_validate */
+struct amdgpu_vm_parser {
+	uint32_t        domain;
+	bool            wait;
+};
+
 
 static inline struct amdgpu_device *get_amdgpu_device(struct kgd_dev *kgd)
 {
@@ -262,6 +268,13 @@ static int amdgpu_amdkfd_bo_validate(struct amdgpu_bo *bo, uint32_t domain,
 	return ret;
 }
 
+static int amdgpu_amdkfd_validate(void *param, struct amdgpu_bo *bo)
+{
+	struct amdgpu_vm_parser *p = param;
+
+	return amdgpu_amdkfd_bo_validate(bo, p->domain, p->wait);
+}
+
 static int amdgpu_amdkfd_bo_invalidate(struct amdgpu_bo *bo)
 {
 	int ret = 0;
-- 
2.7.4

